src/main/java/com/example/MachineService/controllers/TaskController.java

```diff
@@ -0,0 +1,105 @@
+package com.example.MachineService.controllers;
+
+import com.example.MachineService.entities.Machine;
+import com.example.MachineService.entities.Task;
+import com.example.MachineService.services.TaskService;
+import org.hibernate.procedure.ParameterMisuseException;
+import org.hibernate.procedure.ParameterStrategyException;
+import org.springframework.web.bind.annotation.*;
+
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.core.Response;
+import java.util.*;
+
+@RequestMapping("/")
+@RestController
+public class TaskController {
+
+    private final TaskService taskService;
+
+    public TaskController(TaskService taskService) {
+        this.taskService = taskService;
+    }
+
+    @PostMapping("user/{userId}/job/task/")
+    Response addTasks(@RequestBody Task[] tasksArray, @PathVariable Long userId) {
+        try {
+            //checking if there is a conflict with the parameters
+            List<Task> tasks = Arrays.asList(tasksArray);
+            Map<String, String> map= new HashMap<>();
+            for (Task task : tasks) {
+                for (Machine machine: task.getMachines()) {
+                    if(map.containsKey("t"+task.getName()+"m"+machine.getId())) throw new ParameterStrategyException("conflict in input tasks!");
+                    map.put("t"+task.getName()+"m"+machine.getId(),"");
```

Comment:
I don't mean to do it this way, you should have a map that has a String as key which is the machine id and a list of String for the tasks name of this machine.
this way you are not using the map in the right way!